astropy/coordinates/angles/core.py

```diff
@@ -190,6 +193,10 @@ def __new__(cls, angle, unit=None, dtype=np.inexact, copy=True, **kwargs):
 
     @staticmethod
     def _convert_unit_to_angle_unit(unit):
```

Comment:
Caching would be performant, fully backwards compatible and there would be no need to write out a set of angle units as a module level constant.
```suggestion
    @staticmethod
    @functools.cache
    def _convert_unit_to_angle_unit(unit):
```

Reply:
It's indeed in the same ballpark. Since @mhvk already approved let's ask if he'd have any objections.

Reply:
I like the caching since it keeps the code simple -- good idea!